src/Build/Evaluation/Conditionals/StringExpressionNode.cs

```diff
@@ -141,6 +160,19 @@ internal override void ResetState()
             _cachedExpandedValue = null;
         }
 
+        /// <summary>
+        /// Should this node be treated as an expansion of VisualStudioVersion, rather than
+        /// its literal meaning?
+        /// </summary>
+        /// <remarks>
+        /// Needed to provide a compat shim for numeric/version comparisons
+        /// on MSBuildToolsVersion, which were fine when it was a number
+        /// but now cause the project to throw InvalidProjectException when
+        /// ToolsVersion is "Current". https://github.com/Microsoft/msbuild/issues/4150
+        /// </remarks>
+        private bool ShouldBeTreatedAsVisualStudioVersion() =>
```

Comment:
Should we worry about this being slow?

Reply:
I don't expect a straight string comparison to show up on any hot lists, even with IgnoreCase. It is slower than not having this, of course.

Reply:
@davkean will this show up on the perf traces?

Reply:
I can cache the result of this in a bool if you're that worried.

Reply:
When you look at these sorts of things, factor in that opening the CoreFx solution caused 19,000 evaluations to show the Solution Explorer tree.

Reply:
Caching the result turned out to be handy to scope the quirk down further so the current rev does that.